### DISCUSSION TITLE
Multiple separate lattices in one run

### DISCUSSION THREAD

**Discussion by edbennett** [paboyle/Grid#447] [Q&A] [answered]
To implement parallel tempering and similar algorithms, one wants to have multiple separate lattices running as part of the same global communicator such that parameters can be transferred between them.

The optimal implementation of this would be to have each lattice occupying its own subset of the allocated CPUs/GPUs, to minimise the amount of communication. (E.g. one lattice per node could allow such an algorithm to work efficiently even without a high-performance interconnect.) However, if I understand correctly, if you create multiple `Lattice` objects, then each is parallelised across the entire communicator.

One way to achieve the efficient result could be to split `MPI_COMM_WORLD` into multiple communicators, and then replace `MPI_COMM_WORLD` with one of these communicators before calling `Grid_init()`. (Also keeping a copy of the original global communicator to allow communication between the various lattices.) But this seems a little awkward.

Are there any better recommendations for achieving the result of having separate lattices on separate nodes/CPUs while still allowing communication between them? Thanks!

**Comment by aportelli** [comment, marked as answer]
Hi @edbennett, have you looked at `tests/solver/Test_split_grid.cc`? It sounds to me it illustrates what you want.

**Reply by edbennett** [reply]
Aha, I hadn't seen that—thanks! I'll have a closer read and understand better what it's doing.
